### DISCUSSION TITLE
Where I can find Objectsecret for PayGreenJS?

### DISCUSSION THREAD

**Discussion by EgoitzAB** [PayGreen/documentation#16] [Q&A] [answered]
Hello,
I am having a lot of problems to integrate your platform. It is my first integration, so is normal.
I start getting the Bearer Token, I send request to the paymentorder api and when I want to start with the template that have the PaygreenJS, I can't find the ObjectSecret for the init().
I don't know if I am doing well, I am starting to be exhausted. I am using Django, and I can only apply some API steps, but in the moment that I need to make all it's really hard. The documentation don't help.
If someone can help me, I only need to make a little integration for one E-Shop.
Thanks.

**Comment by EgoitzAB** [comment]
Found, I am starting to be exhausted. Now the paygreenjs ask for the Bearer JWT token, but the documentation don't say nothing about that. I am passing in the init the required data and don't work paygreenjs.init({
            paymentOrderID: "po_xxxxxxxxxxxxxxxxxxx",
            objectSecret: "xxxxxxxxxxxx",
            publicKey: "pk_xxxxxxxxxxx",
            mode: "payment",
            style: {
                input: {
                    base: {
                        color: 'black',
                        fontSize: '18px',
                    },
                },
            }
        });
    });

**Reply by EgoitzAB** [reply]
This are the fields, of course I get the token before, and the data is taken from the response of the createpaymentorder interface.

**Comment by enzobillisPG** [comment, marked as answer]
Hi @EgoitzAB 
Sorry for the difficutlies you have during your integration. PayGreen documentation is often updated, [this documentation](https://developers.paygreen.fr/) is also free to request changes. You can up/down vote a page and add a comment on it to explain us what difficulties you encounter while integrating PayGreen. This help us a lot to improve the documentation.

Good to hear you found how to retrieve the objectSecret, this one is in the response once you create a PaymentOrder. You cannot retrieve it again after this reponse. You're right, this is a caveeat in our documentation, we will update it soon as possible to add this part.

Could you explaine more the last part of your last message ? You are talking about the a JWT Token asked by PayGreenJS. If you are talking about the publicKey, this can be found in the back-office. The documentation talking about this part: https://developers.paygreen.fr/docs/api-key-concepts-authenfication#public-key

Enzo

**Reply by EgoitzAB** [reply]
Hi @enzobillisPG First of all, thanks for the answer and I didn't know that there was an option to send comments, I will use.
I am using it from a Django view, when I pass the payment order response data to PaygreenJS in my template, I get the JWT token missing in the browser console. I don't understand why, since you don't have to use it any more than in the previous steps. Tomorrow I will continue with the integration, I don't know if it is because it is used from the Django development server, but I am close :D

**Reply by enzobillisPG** [reply]
Hi,
Could you please share us a screen of this error ?
Enzo

**Reply by EgoitzAB** [reply]
Hello,
Yes of course. Finally I made with HostedPages, and I went back in my git log to take the pictures. The problem is that I make the request, the api accept and send another time, I don't now the reason. And I have this two error messages in the seccond requests 
![Captura desde 2023-09-13 15-52-22](https://github.com/PayGreen/documentation/assets/110356620/6fd1a1d8-3e70-4627-b00e-930ad086d7c4)
![Captura desde 2023-09-13 15-52-16](https://github.com/PayGreen/documentation/assets/110356620/b24867d1-dd92-4450-bf3a-cec2a73a921a)
![Captura desde 2023-09-13 15-52-09](https://github.com/PayGreen/documentation/assets/110356620/7abb46ff-bbe2-4ea7-bdd6-f8e588b6e437)
![Captura desde 2023-09-13 15-50-57](https://github.com/PayGreen/documentation/assets/110356620/ad5ec1cc-f2ec-4f2b-88ce-d4f4aae6ac27)
I don't know if I am adding well the pictures, but if you want to see the nasty code behind yo have in my github commit "paygreen improvements" in the pago views.py. Maybe I could be doing something wrong, now I want to finish this part and the complete page in a new clean project, thank you for the help.

**Reply by enzobillisPG** [reply]
Hi,
Thanks for the pictures ! Happy to see you successfully integrated us through HostedPages. This is the easiest way to do. But not the most beautiful if you want to custom the fields.

As I can see in your screenshoots, it seems to be an issue with you public key. Do you pass the public key in the PGJS init object ? Are you sure this one is not revoked in your back office ? Are you sur you are passing the good object secret in PGJS init ?

Have a good day
Enzo
